### PR TITLE
Create Leaderboard Api

### PR DESCRIPTION
**Pull request by arc9693**

### CODE REVIEW

File: core/urls.py [hunk 1]
```diff
@@ -45,6 +50,9 @@
         # Leaderboard
         url(r'^leaderboard/$', views.leaderboard, name='leaderboard'),
 
+        #Leaderboard api
```
Comment (ghoshbishakh) [inline core/urls.py]: try to use a space after # in every comment

File: core/views.py [hunk 2]
```diff
@@ -193,6 +193,20 @@ def leaderboard(request):
...
+def leaderboardApi(request):
+    response_data = []
+    if request.method == "GET":
+        players = map(lambda x: x.player, User.objects.filter(is_staff=False))
+        players = sorted(players, key=lambda a: a.total_value(), reverse=True)
+
+        for player in players:
+            response_data.append({ "name": str(player.user),"value": str(player.total_value())})
+
+    return HttpResponse(json.dumps(response_data),
```
Comment (ghoshbishakh) [inline core/views.py]: better use [JsonResponse](https://docs.djangoproject.com/en/dev/ref/request-response/#jsonresponse-objects)

**Comment by arc9693** [comment]
@ghoshbishakh please review.

**Comment by ghoshbishakh** [comment]
looks good! I cant test it though. Thanks!

File: core/urls.py [hunk 3]
```diff
@@ -18,6 +20,9 @@
 
         # Stock price JSON data
         url(r'view/stockprice/$', views.stockPrices, name='stockPrice'),
+
+        # Leaderboard api
+        url(r'^api/leaderboard/$', views.leaderboardApi, name='leaderboardApi'),
```
Comment (ankan17) [inline core/urls.py]: This is not required. We don't need to expose the leaderboard api before the event starts.

File: core/views.py [hunk 4]
```diff
@@ -193,6 +193,19 @@ def leaderboard(request):
...
+# Get leaderboard data
+
+def leaderboardApi(request):
+    response_data = []
+    if request.method == "GET":
+        players = User.objects.filter(is_staff=False)
+        players = sorted(players, key=lambda a: a.player.total_value(), reverse=True)
+
+        for player in players:
+            response_data.append({ "name": str(player.username),"value": str(player.player.total_value()),"email": str(player.email)})
```
Comment (ankan17) [inline core/views.py]: I don't think the str method is necessary. Change this line like:
```
response_data.append({
    "name": player.username,
    "value": player.player.total_value(),
    "email": player.email
})
```

**Review by ankan17 — CHANGES_REQUESTED** [review]

File: core/urls.py (hunk 3 above)
Comment (ghoshbishakh) [reply core/urls.py]: views should have a database check which contains their state.. that is enabled or disabled. accordingly the views should return the result or 404. I dont know any cleaner way to enable and disable views.

**Pull request merged**